webapp/src/components/jira_ticket_tooltip/jira_ticket_tooltip.tsx

```diff
@@ -0,0 +1,279 @@
+        }
+
+        if (ticketID === '' && this.props.href.includes('atlassian.net/browse')) {
+            ticketID = this.props.href.split('|')[0].split('?')[0].split('/browse/')[1];
+        }
+
+        this.state = {
+            href: this.props.href,
+            isLoaded: false,
+            ticketId: ticketID,
+        };
+    }
+
+    async init() {
+        let instanceID = '';
+        if (this.props.connectedInstances.length === 1) {
+            instanceID = this.props.connectedInstances[0].instance_id;
+        } else if (this.props.defaultUserInstanceID) {
+            instanceID = this.props.defaultUserInstanceID;
+        }
+
+        let ticketID = '';
+        if (this.props.href.includes('selectedIssue')) {
+            ticketID = this.props.href.split('selectedIssue=')[1].split('&')[0];
+            this.props.getIssueByKey(ticketID, instanceID);
+        }
+
+        if (ticketID === '' && this.props.href.includes('atlassian.net/browse')) {
+            ticketID = this.props.href.split('|')[0].split('?')[0].split('/browse/')[1];
+            if (ticketID) {
+                this.props.getIssueByKey(ticketID, instanceID);
+            }
+        }
+    }
+
+    componentDidMount() {
+        if (this.props.connected && !this.state.isLoaded && ((this.props.ticketDetails && this.props.ticketDetails.ticketId !== this.state.ticketId) || !this.props.isLoaded)) {
+            this.init();
+        } else if (this.props.connected && !this.state.isLoaded && this.props.ticketDetails && this.props.ticketDetails.ticketId === this.state.ticketId) {
+            this.setTicket(this.props);
+        }
+    }
+
+    componentDidUpdate(): void {
+        if (this.props.isLoaded && !this.state.isLoaded && this.props.ticketDetails && this.props.ticketDetails.ticketId === this.state.ticketId) {
+            this.setTicket(this.props);
+        }
+    }
+
+    setTicket(data: Props): void{
+        this.setState({
+            isLoaded: true,
+            ticketDetails: data.ticketDetails,
+        });
+    }
+
+    fixVersionLabel(fixVersion: string) {
+        if (fixVersion) {
+            const fixVersionString = 'Fix Version :';
+            return (
+                <div className='fix-version-label'>
+                    {fixVersionString}
+                    <span className='fix-version-label-value'>
+                        {fixVersion}
+                    </span>
+                </div>
+            );
+        }
+        return (<span/>);
+    }
+
+    tagTicketStatus(ticketStatus: string) {
+        if (ticketStatus.toLowerCase() === 'indeterminate') {
+            return (<span className='ticket-status-indeterminate default-style'>{ticketStatus}</span>);
+        }
+
+        if (ticketStatus.toLowerCase() === 'done') {
+            return (<span className='ticket-status-done default-style'>{ticketStatus}</span>);
+        }
+
+        return (<span className='ticket-status-default default-style'>{ticketStatus}</span>);
```

Comment:
Create a variable for className. Use switch case and decide the value of the variable. At the end, return the `<span>` with the className variable.